nomad/structs/config/consul.go

```diff
@@ -131,12 +133,42 @@ type ConsulConfig struct {
 	// and register with them
 	ClientAutoJoin *bool `mapstructure:"client_auto_join"`
 
-	// ExtraKeysHCL is used by hcl to surface unexpected keys
-	ExtraKeysHCL []string `mapstructure:",unusedKeys" json:"-"`
-
 	// Namespace sets the Consul namespace used for all calls against the
 	// Consul API. If this is unset, then Nomad does not specify a consul namespace.
 	Namespace string `mapstructure:"namespace"`
+
+	// UseIdentity tells the server to sign identities for Consul. In Nomad 1.9+ this
+	// field will be ignored (and treated as though it were set to true).
+	//
+	// UseIdentity is set on the server.
+	UseIdentity *bool `mapstructure:"use_identity"`
+
+	// ServiceIdentity is intended to reduce overhead for jobspec authors and make
+	// for graceful upgrades without forcing rewrite of all jobspecs. If set, when a
+	// job has a service block with the “consul” provider, the Nomad server will sign
+	// a Workload Identity for that service and add it to the service block. The
+	// client will use this identity rather than the client's Consul token for the
+	// group_service and envoy_bootstrap_hook.
+	//
+	// The name field of the identity is always set to
+	// "consul-service/${service_name}-${service_port}".
+	//
+	// ServiceIdentity is set on the server.
+	ServiceIdentity *WorkloadIdentity `mapstructure:"service_identity"`
+
+	// TemplateIdentity is intended to reduce overhead for jobspec authors and make
+	// for graceful upgrades without forcing rewrite of all jobspecs. If set, when a
+	// job has both a template block and a consul block, the Nomad server will sign a
+	// Workload Identity for that task. The client will use this identity rather than
+	// the client's Consul token for the template hook.
+	//
+	// The name field of the identity is always set to "consul".
+	//
+	// TemplateIdentity is set on the server.
+	TemplateIdentity *WorkloadIdentity `mapstructure:"template_identity"`
```

Comment:
I wonder if it would be good to group the fields per where they should be set, like

```hcl
// Client configuration values.
...

// Server configuration values
...
```

Or maybe even different structs? Though that could require quite a bit of adjustments. But having to look at individual field docs can be confusing.